libs/librepcb/core/project/board/drc/boarddesignrulecheck.cpp

```diff
@@ -2284,6 +2301,76 @@ QVector<Path> BoardDesignRuleCheck::getViaLocation(
   return {Path::circle(via.size).translated(via.position)};
 }
 
+bool BoardDesignRuleCheck::isViaUseless(const Data& data,
+                                        const Data::Via& via) noexcept {
+  // The layers of the traces directly connected to the via have already been
+  // added, so if there are two or more connected layers, the via is not
+  // useless and we can skip all the other checks.
+  if (via.connectedLayers.count() >= 2) {
+    return false;
+  }
+
+  QSet<const Layer*> connectedLayers(via.connectedLayers);
+  const ClipperLib::IntPoint viaPosition =
+      ClipperHelpers::convert(via.position);
+
+  // A via is considered to be connected to a plane if it contains
+  // the layer the plane is on, both of them are part of a net, they are in
+  // the same net, and the via is physically within the plane
+  for (const Data::Plane& plane : data.planes) {
+    const int copperNumber = plane.layer->getCopperNumber();
+    if (via.startLayer->getCopperNumber() > copperNumber ||
+        via.endLayer->getCopperNumber() < copperNumber || !plane.netSignal ||
+        !via.netSignal || plane.netSignal != via.netSignal ||
+        connectedLayers.contains(plane.layer)) {
```

Comment:
If every condition contained parenthesis, I'd sleep better :see_no_evil: I've seen too many nasty bugs due to unexpected operator precedence so I really prefer to make it explicit (IMHO it's even more readable). Just a note, I'll accept it as-is.

(same applies to the condition below)